envutil/envutil.go

```diff
@@ -41,3 +44,34 @@ func GetenvWithDefault(envKey, defValue string) string {
 	}
 	return retVal
 }
+
+// RequiredEnv - returns the env's value if specified, otherwise it returns an error that explains the key is required.
+// Use this function to reduce error prone code duplication.
+// E.g. instead of doing this in your code:
+//
+// ```
+// myVar1 := os.Getenv("MY_ENV1")
+// if len(myVar1) < 1 {
+// 	return nil, errors.New("MY_ENV1 required")
+// }
+// ```
+//
+// You can use this function like:
+//
+// ```
+// myVar1, err := requiredEnv("MY_ENV1")
+// if err != nil {
+// 	return nil, errors.WithStack(err)
+// }
+// ```
+//
+// In the first example you have to specify myVar1 and MY_ENV1 two times, which can lead to
+// issues if you copy paste that code but e.g. forget to change the var name in the `if len(myVar1) < 1` line,
+// or you might simply forget to change the var key in the error.
+func RequiredEnv(envKey string) (string, error) {
```

Comment:
If you change to the fast-exit way of doing this then you can pull the value assignment into the if for example:
```
func RequiredEnv(envKey string) (string, error) {
	if val := os.Getenv(envKey); len(val) > 0 {
		return val, nil
	}
	return "", fmt.Errorf("Required environment variable (%s) not provided", envKey)
}
```